addons/udes_stock/models/stock_picking.py

```diff
@@ -10,6 +10,116 @@ class StockPicking(models.Model):
         "Sequence", default=0, help="Used to order the 'All Operations' kanban view"
     )
 
+    # Related pickings computed fields
+
+    u_first_picking_ids = fields.One2many(
+        "stock.picking",
+        string="First Pickings",
+        compute="_compute_first_picking_ids",
+        help="First pickings in the chain",
+    )
+    u_prev_picking_ids = fields.One2many(
+        "stock.picking",
+        string="Previous Pickings",
+        compute="_compute_related_picking_ids",
+        help="Previous pickings",
+    )
+    u_next_picking_ids = fields.One2many(
+        "stock.picking",
+        string="Next Pickings",
+        compute="_compute_related_picking_ids",
+        help="Next pickings",
+    )
+    u_created_backorder_ids = fields.One2many(
+        "stock.picking",
+        string="Created Backorders",
+        compute="_compute_related_picking_ids",
+        help="Backorders created from this picking",
+    )
+
+    # Picking quantity computed fields
+
+    u_quantity_done = fields.Float(
+        "Quantity Done",
+        compute="_compute_picking_quantities",
+        digits=(0, 0),
+        help="Quantity done of the moves related to the picking",
+    )
+    u_total_quantity = fields.Float(
+        "Total Quantity",
+        compute="_compute_picking_quantities",
+        digits=(0, 0),
+        help="Total quantity todo of the moves related to the picking",
+    )
+    u_has_discrepancies = fields.Boolean(
+        "Has Discrepancies",
+        compute="_compute_picking_quantities",
+        readonly=True,
+        help="Flag to indicate if the picking has discrepancies.",
+    )
+    u_num_pallets = fields.Integer(
+        "Total Pallets Count",
+        compute="_compute_num_pallets",
+        help="Total number of different pallets in the picking",
+    )
+
+    @api.depends("move_lines", "move_lines.move_orig_ids", "move_lines.move_orig_ids.picking_id")
+    def _compute_first_picking_ids(self):
+        Move = self.env["stock.move"]
+
+        for picking in self:
+            first_moves = Move.browse()
+
+            moves = picking.move_lines
+            while moves:
+                first_moves |= moves.filtered(lambda x: not x.move_orig_ids)
+                moves = moves.move_orig_ids
+
+            picking.u_first_picking_ids = first_moves.picking_id
+
+    @api.depends(
+        "move_lines",
+        "move_lines.move_orig_ids",
+        "move_lines.move_dest_ids",
+        "move_lines.move_orig_ids.picking_id",
+        "move_lines.move_dest_ids.picking_id",
+    )
+    def _compute_related_picking_ids(self):
+        """ Compute previous/next picking """
+        for picking in self:
+            picking.u_prev_picking_ids = picking.move_lines.move_orig_ids.picking_id
+            picking.u_next_picking_ids = picking.move_lines.move_dest_ids.picking_id
+
+            picking.u_created_backorder_ids = self.search([("backorder_id", "=", picking.id)])
```

Comment:
In 11 there is check for picking.id , is it not needed?

Reply:
No, because computed fields are only run on saved records, so the record will always have an ID set.